### DISCUSSION TITLE
fwupdmgr refresh on a Fitlet device can't resolve hostname

### DISCUSSION THREAD

**Discussion by carlospenedorocha** [fwupd/fwupd#2863] [Q&A]
**Describe the bug**
First, let me say that I'm not totally sure this issue is related to fwupd or to the firmware vendor itself or LVFS per se. 

I have a Fitlet device (from Compulab) running Ubuntu Core16. All snaps are up to date.

When I run `sudo uefi-fw-tools.fwupdmgr refresh`, I get the following:
"Failed to download https://secure-lvfs.rhcloud.com/downloads/firmware.xml.gz.asc: Cannot resolve hostname"

From the device, I can ping any other hosts, even "rhcloud.com". It's not a DNS thing going on in the device.

**Steps to Reproduce**
1. on a Fitlet device, just run `uefi-fw-tools.fwupdmgr refresh`

**Expected behavior**
Should be able to refresh.

**fwupd version information**

It is a snap package.
_uefi-fw-tools: version 1.5.4-0.7.2+git, rev 48, latest/beta_ (same issue was observed in rev 22 latest/stable)

**fwupd device information**
![image](https://user-images.githubusercontent.com/10661562/107507644-27b45080-6ba0-11eb-9152-edd25f8fb2fa.png)

**System UEFI configuration**
Please provide the output of the following commands:
```shell
$ efibootmgr -v
```
![image](https://user-images.githubusercontent.com/10661562/107531967-abc90100-6bbd-11eb-8e28-8fdf1607f377.png)

**Additional questions**
- Operating system and version: _Ubuntu Core16_
- Have you tried rebooting? _Yes_
- Is this a regression? _Don't know, first time trying, but I know it works on Dell gateway devices_ 
- Are you using an NVMe disk? _Not sure, probably not as per the specs_
- Is secure boot enabled? _No_
- Is this a Lenovo system with 'Boot Order Lock' turned on in the BIOS? _No_

**Comment by hughsie** [comment]
> Failed to download https://secure-lvfs.rhcloud.com/downloads/firmware.xml.gz.asc

We've not used that domain for several years. I think you've installed the wrong snap.

**Reply by emirozer** [reply]
@hughsie Hi, we have the uefi-fw-tools snap latest/beta channel. check the screenshot @carlospenedorocha posted

**Comment by superm1** [comment]
@woodrow-shen can you please help direct @carlospenedorocha to the correct place to get the `uefi-fw-tools` snap fixed?

**Reply by carlospenedorocha** [reply]
@hughsie I just found that URL (secure-lvfs....) in /snap/uefi-fw-tools/current/etc/fwupd.conf. That URL is wrong in uefi-fw-tools latest/beta (rev 48 at least). Once I installed latest/stable, the problem I reported is gone (of course the URL in stable is the correct one). See below the wrong one:

![image](https://user-images.githubusercontent.com/10661562/107621989-6b1dc600-6c57-11eb-93e8-4a2e65902a3b.png)

**Reply by woodrow-shen** [reply]
@carlospenedorocha Unfortunately we stop maintaining the version 1.5.4-0.7.2+git from latest track, and you can use `sudo snap install uefi-fw-tools --channel=18/edge` to see if it can help.

**Comment by osteenbergen** [comment]
Using `snap download uefi-fw-tools --beta` you can extract `/snap/snapcraft.yaml` and see the steps used to produce the package.

Important snippet:
```yaml
fwupd:
    plugin: autotools
    source: https://github.com/timchen119/fwupd.git
    source-type: git
    source-branch: uefi-fw-tools-uc16-snap
    build-packages:
      - docbook-utils
      - gettext
      - intltool
```

Clearly the package published to edge/beta snap channel is using a forked and outdated repository.

**Reply by hughsie** [reply]
We've never built the uefi-fw-tools it's completely different from the official snap package: https://snapcraft.io/fwupd

**Comment by woodrow-shen** [comment]
@carlospenedorocha As you're talking about uefi-fw-tools instead of fwupd snap, I will suggest to file an issue to https://github.com/Lyoncore/fwupd-snapcraft regarding uefi-fw-tools snap.
